serving/src/main/java/ai/djl/serving/http/ManagementRequestHandler.java

```diff
@@ -242,40 +213,43 @@ private void handleRegisterModel(final ChannelHandlerContext ctx, QueryStringDec
                                 v -> {
                                     for (ModelInfo<Input, Output> m : workflow.getModels()) {
                                         modelManager.initWorkers(
-                                                m, deviceName, minWorkers, maxWorkers);
+                                                m,
+                                                req.getDeviceName(),
+                                                req.getMinWorkers(),
+                                                req.getMaxWorkers());
                                     }
                                 })
                         .exceptionally(
                                 t -> {
                                     NettyUtils.sendError(ctx, t.getCause());
-                                    if (synchronous) {
+                                    if (req.isSynchronous()) {
                                         String name = workflow.getName();
-                                        modelManager.unregisterWorkflow(name, version);
+                                        modelManager.unregisterWorkflow(name, req.getVersion());
                                     }
                                     return null;
                                 });
-        if (synchronous) {
-            final String msg = "Model \"" + modelName + "\" registered.";
+        if (req.isSynchronous()) {
+            final String msg = "Model \"" + req.getModelName() + "\" registered.";
             f.thenAccept(v -> NettyUtils.sendJsonResponse(ctx, new StatusResponse(msg)));
         } else {
-            String msg = "Model \"" + modelName + "\" registration scheduled.";
+            String msg = "Model \"" + req.getModelName() + "\" registration scheduled.";
```

Comment:
Is msg supposed to be final in one block and not the other?

Reply:
When a local variable used in inner class (Lambda), it must be final (or effective final). We explicitly add final to indicate it's being used in inner class